### ISSUE TITLE
Textarea: autoResize doesn't work when used inside a Dialog

### ISSUE THREAD

**Issue by Timmeeeey** [primefaces/primeng#9231]
**I'm submitting a ...**
```
[x] bug report
[ ] feature request
[ ] support request
```

**Plunkr Case (Bug Reports)**
https://stackblitz.com/edit/github-fggov7

**Current behavior**
When a textarea with autoResize is used inside a dialog the height of the textarea is initially set to 0px.
It doesn't matter if the textarea is empty or not.

![image](https://user-images.githubusercontent.com/46870283/92106253-6a4d5200-ede4-11ea-883b-fe1ba05c6a4d.png)
![image](https://user-images.githubusercontent.com/46870283/92106714-14c57500-ede5-11ea-8bed-78871f9e70a3.png)

When you click inside the textarea the height is set correctly.

![image](https://user-images.githubusercontent.com/46870283/92106298-7afdc800-ede4-11ea-8e7c-f165d98c4abb.png)
![image](https://user-images.githubusercontent.com/46870283/92106756-2149cd80-ede5-11ea-92d4-5a17984f83c9.png)

The problem appeared since #9129 was fixed.

**Expected behavior**
The height of a textarea with autoResize should be calculated correctly.

**Minimal reproduction of the problem with instructions**
```
<button pButton (click)="displayDialog = true" label="Open Dialog"></button>

<p-dialog [(visible)]="displayDialog">
  <textarea pInputTextarea rows="3" autoResize="autoResize"></textarea>
</p-dialog>
```

* **Angular version:** 10

* **PrimeNG version:** 10.0.2

* **Browser:** Tested in Chrome, Firefox

**Comment by serega270** [comment]
@Timmeeeey 
This seems to happen because the resize occurs when the `p-dialog` is not even opened
Try to add `*ngIf="displayDialog"` to the component

```
<p-dialog [(visible)]="displayDialog" *ngIf="displayDialog">
...
</p-dialog>
```

**Comment by Timmeeeey** [comment]
@serega270 
Thanks, this seems to work but I think it shouldn't be necessary to have `visible` and `*ngIf` at the same time.
Also I have to change all my dialogs.
I found out that wrapping the `this.resize()` in `ngAfterViewInit` inside a `setTimeout` also solves the problem.

```
InputTextarea.prototype.ngAfterViewInit = function() {
  if (this.autoResize) {
    window.setTimeout(() => this.resize());
  }
};
```

**Comment by testowy-uzytkownik-1** [comment]
Related to #7000

**Comment by cetincakiroglu** [comment]
Hi,

We couldn't replicate the issue with the latest PrimeNG (v13).
If the issue still exists with the latest PrimeNG version, please provide a StackBlitz example by using [PrimeNG Issue Template ](https://stackblitz.com/github/primefaces/primeng-issue-template) and re-open the issue.

Regards

**Comment by Timmeeeey** [comment]
@cetincakiroglu Here is the updated version:
https://stackblitz.com/edit/github-7k18kj

Unfortunately I cannot repoen this issue. Could you please reopen it?

![2022-03-29_08h40_05](https://user-images.githubusercontent.com/46870283/160549601-05c6027a-45e7-4ae4-a46f-756d9a075cfd.gif)

**Comment by Cito** [comment]
Please allow me to express my frustration about how bug reports are ignored for many months and then suddenly closed as "not reproducable with latest version" even when nothing has been fixed or changed in that regard. At least give us some time to react before closing the issue, since even the original author cannot reopen an issue that has been closed by an admin. The same happened to me with another issue yesterday.

**Comment by dev4ndy** [comment]
I have the same problem with primeng-lts 12.2.5 version, do you have any plan to fix it? it appears that this error has been there since 2018: https://github.com/primefaces/primeng/issues/7000

**Comment by dev4ndy** [comment]
> Hi,
> 
> We couldn't replicate the issue with the latest PrimeNG (v13). If the issue still exists with the latest PrimeNG version, please provide a StackBlitz example by using [PrimeNG Issue Template ](https://stackblitz.com/github/primefaces/primeng-issue-template) and re-open the issue.
> 
> Regards

@cetincakiroglu so the only way to solve this is to upgrade?

**Comment by Timmeeeey** [comment]
The bug still exists in PrimeNg 14.2.2.

@cetincakiroglu Could you please reopen this issue?

https://stackblitz.com/edit/github-7k18kj-hivu3g

**Comment by ProfEibe** [comment]
@cetincakiroglu It is still there in PrimeNG 16

**Comment by cetincakiroglu** [comment]
Fixed now, it'll be available in the upcoming release.

Thanks for reporting the issue!

**Comment by Emiliorth** [comment]
Still exists in 17.8

**Comment by Taleynikov** [comment]
> Still exists in 17.8, now even with ngIf

I solved the problem "this way"

![image](https://github.com/primefaces/primeng/assets/28928134/113e3702-3697-4ac3-a192-a6aeed5d0975)
![image](https://github.com/primefaces/primeng/assets/28928134/d38d198b-881e-4094-b66d-7496b69dee09)

**Comment by Emiliorth** [comment]
Ah yea, me either, maybe with setTimeout

**Comment by jules-js** [comment]
issue still exists in 17.12.0!

**Comment by Junior-Stalin** [comment]
still exists in 17.18.6

**Comment by marcschroeder** [comment]
The issue seems to happen when using a Textarea in a Tabview as well. It only seems to be an issue when using the autoResize option.

**Comment by marcschroeder** [comment]
@cetincakiroglu can we reopen this issue?

**Comment by WStewart110** [comment]
Also having this issue on tabView.

**Comment by vanelli26** [comment]
[html]
<textarea
  id="dsObservacao"
  pInputTextarea
  formControlName="dsObservacao"
  placeholder="Observação"
  rows="3"
  cols="30"
  [autoResize]="true">
</textarea>

[css]
textarea#dsObservacao {
  height: auto !important;
}

**Comment by AnthonyLzq** [comment]
Guys, this problem persist, why are you closing something that is not solved yet?

In my case it's happening when there is a `textarea` inside a `p-tabView`. I'm using `v^15.4.1`.

I was able to "solved it" using a `min-height` attribute in my styles. In my case, the `textarea` inside the `p-tabView` renders properly, so I got the `height` from there and then I used it as follows:

```html
<textarea
  rows="12"
  pInputTextarea
  [autoResize]="true"
  style="resize: none; min-height: 200px !important"
></textarea>
```

Hope it helps.
